### ISSUE TITLE
com.hazelcast.jet.MulticastDiscoveryTest.when_twoJetInstancesCreated_then_clusterOfTwoShouldBeFormed

### ISSUE THREAD

**Issue by olukas** [hazelcast/hazelcast-jet#1503]
_master_  (commit 8ceac329410dcc9bfeaade4281e4e18823762152)

Failed on Sonar: http://jenkins.hazelcast.com/job/jet-oss-master-sonar-nightly/192/testReport/junit/com.hazelcast.jet/MulticastDiscoveryTest/when_twoJetInstancesCreated_then_clusterOfTwoShouldBeFormed/

Stacktrace:
```
java.lang.AssertionError: expected:<2> but was:<3>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.jet.MulticastDiscoveryTest.when_twoJetInstancesCreated_then_clusterOfTwoShouldBeFormed(MulticastDiscoveryTest.java:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
Hiccups measured while running test 'when_twoJetInstancesCreated_then_clusterOfTwoShouldBeFormed(com.hazelcast.jet.MulticastDiscoveryTest):'
13:08:30, accumulated pauses: 98 ms, max pause: 57 ms, pauses over 1000 ms: 0
13:08:35, accumulated pauses: 236 ms, max pause: 219 ms, pauses over 1000 ms: 0
```

**Comment by viliam-durina** [comment]
This test creates real-network instances. It likely formed a cluster with another instance running in the same JVM. We can't rewrite it to use mock-networking, as it is designed to test multicast discovery. 

The third instance was running in another JVM, the test logs contain only entries for members on ports 5702 and 5703, the third member had 5701 port.

**Comment by viliam-durina** [comment]
Fixed in #1562. Not this is the only test that uses real-network instances, so should not happen again.
